frontend/src/App.tsx

```diff
@@ -19,17 +19,42 @@ import { useEffect, useState } from 'react'
 import { User, onAuthStateChanged } from 'firebase/auth'
 import { AuthProvider, PrivateRoute, PublicRoute } from '@auth'
 import { auth } from '@fire'
+import axios from 'axios'
 
 const App = () => {
   const [currentUser, setCurrentUser] = useState<User | null>(null)
   const [isLoading, setIsLoading] = useState(true)
+  const [token, setToken] = useState('')
 
   useEffect(() => {
     onAuthStateChanged(auth, (user) => {
       setCurrentUser(user)
-      setIsLoading(false)
+      // need these to conditions to resolve isLoading at the correct time so data can be loaded properly
+      if (user)
+        user
+          .getIdToken(false)
+          .then((idToken) => {
+            // store token in local storage
+            axios.interceptors.request.use(
+              (request) => {
+                request.headers.authtoken = idToken
+                return request
+              },
+              (error) => {
+                return Promise.reject(error)
+              }
+            )
+            setToken(idToken) // might have to leave this in here
+            setIsLoading(false)
+          })
+          .catch(() => {
+            setIsLoading(false)
+          })
+      else {
+        setIsLoading(false)
+      }
     })
-  }, [])
+  }, [currentUser, token])
```

Comment:
I thought we left it as `[]` because onAuthChanged binds a callback function that runs whenever auth state changes already?

Reply:
Initially, for some reason, the first login would always not be authorized (return a 403), and I wasn't sure why. I tried a bunch of random combinations and this one was the only one that worked consistently, but that being said, it is one of said random combinations, so I guess I can start process of elimination to see which parts are actually needed.